kb/all-the-ways-to-run-a-cluster.md

```diff
@@ -0,0 +1,123 @@
+
+## Docker & Docker Compose
+
+You can easily run a Temporal Cluster in [Docker](https://docs.docker.com/engine/install) containers using [Docker Compose](https://docs.docker.com/compose/install).
+
+If you have Docker and Docker Compose installed, all you need to do is clone the [temporalio/docker-compose](https://github.com/temporalio/docker-compose) repo and run the `docker-compose up` command from its root.
+
+The `temporalio/docker-compose` repo comes loaded with a variety of configuration templates that enable you to try all three databases that the Temporal Platform supports (PostgreSQL, MySQL, Cassandra).
+It also enables you to try [Advanced Visibility](/visibility/#advanced-visibility) using [Search Attributes](/visibility/#search-attribute), emit metrics, and even play with the [Archival](/clusters/#archival) feature.
+
+Running your Cluster in Docker is convenient and enables you to play with features.
+However, it does not offer the same performace in terms of processing Workflow Executions per second as Temporalite.
+Granted, you would notice this limitation only if you run hundreds of Workflows concurrently.
+
+The following commands start and run a Temporal Cluster in Docker using the default configuration (docker-compose.yml):
+
+```bash
+git clone https://github.com/temporalio/docker-compose.git
+cd docker-compose
+docker-compose up
+```
+
+Local [Temporal Clients](/temporal/#temporal-client) and [Workers](/workers) can connect to the Cluster running in Docker at 127.0.0.1:7233 (default connection for most SDKs) and the Temporal Web UI at 127.0.0.1:8080.
+
+To try other configurations (different dependencies and databases), or to try a custom Docker image, follow the [temporalio/docker-compose README](https://github.com/temporalio/docker-compose/blob/main/README.md).
+
+## Import the Server package
+
+The Temporal Server is a standalone Go application that can be [imported](/references/server-options) into another project.
+
+The main reason you might want to do this is to pass in custom plugins or any other customizations through the [Server Options](/references/server-options).
+Then you can build and run a binary that contains your customizations.
+
+Doing this requires [Go v1.16+](https://github.com/temporalio/temporal/blob/master/CONTRIBUTING.md).
+
+## Temporal Server as a binary
+
+You can run the Temporal Server as a single Go binary, or you can run each service within the Server separately.
+For example, if you are using Kubernetes, you could have one service per pod so they can be scaled independently in future.
+
+In Docker, you could run each service in its own container, using the `SERVICES` flag to specify the service:
+
+```bash
+docker run
+    # persistence/schema setup flags omitted
+    -e SERVICES=history \                      -- Spin up one or more: history, matching, worker, frontend
```

Comment:
would be great to give example for multiple roles. Is it space/comma separated list?